crates/runtime/src/dataupdate.rs

```diff
@@ -75,6 +78,24 @@ impl StreamingDataUpdate {
     }
 }
 
+impl TryFrom<DataUpdate> for StreamingDataUpdate {
+    type Error = DataFusionError;
+
+    fn try_from(data_update: DataUpdate) -> std::result::Result<Self, Self::Error> {
+        let schema = Arc::clone(&data_update.schema);
+        let data = Box::pin(MemoryStream::try_new(
+            data_update.data,
+            data_update.schema,
+            None,
+        )?) as SendableRecordBatchStream;
+        Ok(Self {
+            schema,
+            data,
+            update_type: data_update.update_type,
+        })
+    }
+}
+
 pub struct DataUpdateExecutionPlan {
```

Comment:
Is anything still using this?

Reply:
Yes, it it is still used in [datafusion.write_data](https://github.com/spiceai/spiceai/blob/0839f73a8e646aae38fea02c7dd9cbd577c098d9/crates/runtime/src/datafusion.rs#L432) (used by `oss_benchmarks`, `query_history`, `opentelemetry`, `spice_metrics`) and also in [changes stream](https://github.com/spiceai/spiceai/blob/0839f73a8e646aae38fea02c7dd9cbd577c098d9/crates/runtime/src/accelerated_table/refresh_task/changes.rs#L168).

Do we want to update all such places so we have single method to write data to accelerator?

Reply:
Yes, but it can be in a separate PR

Reply:
Noted  👍   - confirm this will be updated next.